pkg/flow/adapter/xslttransform/adapter_test.go

```diff
@@ -256,6 +257,67 @@ func TestXSLTTransformEvents(t *testing.T) {
 	}
 }
 
+func TestXSLTTransformKSINK(t *testing.T) {
+	testCases := map[string]struct {
+		allowXSLTOverride bool
+		xslt              string
+
+		inEvent             cloudevents.Event
+		expectedEventString string
+		expectPanic         string
+		expectCategory      string
+	}{
+		"transform ok": {
+			xslt:                tXSLT,
+			inEvent:             newCloudEvent(tXML, cloudevents.ApplicationXML),
+			expectedEventString: "Context Attributes,\n  specversion: 1.0\n  type: ce.test.type\n  source: ce.test.source\n  id: ce-abcd-0123\n  datacontenttype: application/xml\nData (binary),\n  <?xml version=\"1.0\"?>\n<output>\n  <item>A1</item>\n  <item>B2</item>\n  <item>C3</item>\n</output>\n\n",
+			expectCategory:      tSuccessAttribute,
+		},
+	}
+	for name, tc := range testCases {
+		t.Run(name, func(t *testing.T) {
+			ceClient := adaptertest.NewTestClient()
+
+			env := &envAccessor{
+				EnvConfig: adapter.EnvConfig{
+					Component: tCloudEventSource,
+				},
+				BridgeIdentifier: tBridgeID,
+			}
+
+			ctx := context.Background()
+			style, err := parseXSLT(tc.xslt)
+			assert.NoError(t, err)
+
+			replier, err := targetce.New(env.Component, logtesting.TestLogger(t),
+				targetce.ReplierWithStatefulHeaders(env.BridgeIdentifier),
+				targetce.ReplierWithStaticDataContentType(cloudevents.ApplicationXML),
+				targetce.ReplierWithStaticErrorDataContentType(*cloudevents.StringOfApplicationJSON()),
+				targetce.ReplierWithPayloadPolicy(targetce.PayloadPolicy(targetce.PayloadPolicyAlways)),
+				targetce.ReplierWithStaticResponseType(v1alpha1.EventTypeXSLTTransformError))
+			assert.NoError(t, err)
+
+			a := &xsltTransformAdapter{
+				logger: logtesting.TestLogger(t),
+
+				ceClient:     ceClient,
+				xsltOverride: false,
+				defaultXSLT:  style,
+				replier:      replier,
+				sink:         "http://localhost:8080",
+			}
+
+			_, r := a.Dispatch(ctx, tc.inEvent)
```

Comment:
Aren't we going to check the response while we're at it?

Reply:
yea i tried.. but i failed... it is supposed to return this:
```
*event.Event((*event.Event)(nil))
```

But i don't know how to make this to compare it against

Reply:
how can i make this? @antoineco

Reply:
`assert.Nil()` should work

Reply:
got it! that worked.